cmd/generic-handlers.go

```diff
@@ -469,3 +477,90 @@ func (h pathValidityHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	}
 	h.handler.ServeHTTP(w, r)
 }
+
+// logginHandler logs all HTTP requests and responses
+// and write the log to a io.Writer object.
+type loggingHandler struct {
+	writer    io.Writer
+	writeLock *sync.Mutex
+	handler   http.Handler
+}
+
+func (h loggingHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+
+	const timeFormat = "2006-01-02 15:04:05 -0700"
+	const recordLen = 32 * 1024
+
+	// Generate short random request ID
+	reqID := fmt.Sprintf("%X", time.Now().UnixNano()%1e8)
```

Comment:
> I think we are decided to do dmesg style request id.

oh yes.. but then I thought isn't it better to have complete time (full date, time) than seconds since boot time ? (sorry for not bringing this up).